packages/platform-sdk-profiles/src/profiles/profile.ts

```diff
@@ -242,19 +242,21 @@ export class Profile implements ProfileContract {
 	 */
 	public async restore(password?: string): Promise<void> {
 		let data: ProfileStruct | undefined;
+		let errorReason = "";
 
 		try {
 			if (password !== undefined) {
 				data = this.decrypt(password);
 			} else {
 				data = JSON.parse(Base64.decode(this.#data.data));
 			}
-		} catch {
-			// Do nothing if both fail...
+		} catch (e) {
+			// Append the error reason
+			errorReason = ` Reason: ${e.message}`;
 		}
 
 		if (data === undefined) {
-			throw new Error("Failed to decode or decrypt the profile.");
+			throw new Error(`Failed to decode or decrypt the profile.${errorReason}`);
```

Comment:
@faustbrian there could be cases that the reason might be helpful, e.g restoring and decrypting a passwordless profile would error that the password is not required. https://github.com/ArkEcosystem/platform-sdk/pull/858/files#diff-f4f13df0c990dcf1ed7c0271688dfb420d39dbc82a611b044bfc63414b2fb01dR284